mxfusion/components/distributions/wishart.py

```diff
@@ -0,0 +1,264 @@
+import numpy as np
+import mxnet as mx
+
+from mxfusion.common.exceptions import InferenceError
+from ...util import special as sp
+from ...common.config import get_default_MXNet_mode
+from ..variables import Variable
+from .distribution import Distribution, LogPDFDecorator, DrawSamplesDecorator
+from ..variables import is_sampled_array, get_num_samples
+from ...util.customop import broadcast_to_w_samples
+
+
+class WishartLogPDFDecorator(LogPDFDecorator):
+
+    def _wrap_log_pdf_with_broadcast(self, func):
+        def log_pdf_broadcast(self, F, **kw):
+            """
+            Computes the logarithm of the probability density/mass function (PDF/PMF) of the distribution.
+            The inputs and outputs variables are in RTVariable format.
+
+            Shape assumptions:
+            * degrees_of_freedom is S x 1
+            * scale is S x N x D x D
+            * random_variable is S x N x D x D
+
+            Where:
+            * S, the number of samples, is optional. If more than one of the variables has samples,
+            the number of samples in each variable must be the same. S is 1 by default if not a sampled variable.
+            * N is the number of data points. N can be any number of dimensions (N_1, N_2, ...) but must be
+             broadcastable to the shape of random_variable.
+            * D is the dimension of the distribution.
+
+            :param F: the MXNet computation mode (mxnet.symbol or mxnet.ndarray)
+            :param kw: the dict of input and output variables of the distribution
+            :type kw: {str (name): MXNet NDArray or MXNet Symbol}
+            :returns: log pdf of the distribution
+            :rtypes: MXNet NDArray or MXNet Symbol
+            """
+            variables = {name: kw[name] for name, _ in self.inputs}
+            variables['random_variable'] = kw['random_variable']
+            rv_shape = variables['random_variable'].shape[1:]
+
+            num_samples = max([get_num_samples(F, v) for v in variables.values()])
+
+            shapes_map = dict(
+                degrees_of_freedom=(num_samples,),
+                scale=(num_samples,) + rv_shape,
+                random_variable=(num_samples,) + rv_shape
+            )
+            variables = {name: broadcast_to_w_samples(F, v, shapes_map[name])
+                         for name, v in variables.items()}
+            res = func(self, F=F, **variables)
+            return res
+        return log_pdf_broadcast
+
+
+class WishartDrawSamplesDecorator(DrawSamplesDecorator):
+
+    def _wrap_draw_samples_with_broadcast(self, func):
+        def draw_samples_broadcast(self, F, rv_shape, num_samples=1,
+                                   always_return_tuple=False, **kw):
+            """
+            Draw a number of samples from the distribution. The inputs and outputs variables are in RTVariable format.
```

Comment:
```suggestion
            Draw a number of samples from the distribution.
```

I know you didn't write this, but could you change this comment in the other shape decorators to not refer to RTVariables? That's an old concept we got rid of a while back.